### PR TITLE
UI: Show link to release notes in Help menu

### PR DESCRIPTION
**Pull request by WizardCM**
### Description
Dynamically shows a Release Notes link if the `OBS_VERSION` tag exists on GitHub.

image

### Motivation and Context
Currently, users are only exposed to the release notes once - before they update. This provides a quick and easy link that doesn't require digging.

### How Has This Been Tested?
* Build with a valid tag. See the link, click it.
* Build with a non-existent tag or without an internet connection. Don't see the link.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

**Comment by notr1ch** [comment]
I'm not sure I agree with polling github on every startup, this could be added at build time and avoid unnecessary network activity.

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -3205,6 +3206,37 @@ void OBSBasic::TimedCheckForUpdates()
 #endif
 }
 
+void OBSBasic::ShowReleaseNotes()
```
Comment (notr1ch) [inline UI/window-basic-main.cpp]: As this doesn't actually show the release notes, a more appropriate name might be CheckReleaseNotes or FetchReleaseNotes.

**Comment by WizardCM** [comment]
> I'm not sure I agree with polling github on every startup, this could be added at build time and avoid unnecessary network activity.

Yeah, that's another option. Not sure about the best way to do that programmatically though.

**Comment by cg2121** [comment]
I would suggest putting the what's new link in the about dialog. That's where most programs put it.

**Comment by jp9000** [comment]
Submit this again when you figure out how to do it without querying on every startup.

**Comment by jp9000** [comment]
Also you might want to look at the about dialog for an example of how to do that.

**Comment by WizardCM** [comment]
Alrighty, will do!

**Comment by WizardCM** [comment]
Reopening to gather feedback on which method of implementation would be better.

I've updated the branch. Now it **only** makes the network call:
1. **When** the Help menu is opened rather than on startup
2. **Only** the first time the Help menu is opened in a session
3. **If** the OBS_VERSION is manually set

Now, I can think of a few potential ways to improve this, but want to be sure before I go ahead and implement it.

1. [ ] I could switch the current `GetRemoteFile` call to a `RemoteTextThread`, similar to the Patreon list in the About dialog. I _think_ this is worth doing, to ensure it doesn't interrupt normal usage.
2. [ ] I could set a "ReleaseNotesValid" global setting the first time I fetch the URL. The value would be the `OBS_VERSION` and some form of `true/false`. If `OBS_VERSION-true`, it is not fetched again until after an OBS update (where `OBS_VERSION` is mismatched), but the menu item is shown because the version is confirmed valid. `OBS_VERSION-false` would mean no network call and no menu item.
3. [ ] As suggested by cg2121, I could move it somewhere into the about dialog, and so the API call would be made when the dialog opens, rather than when the Help menu opens. I don't mind either way, but I can't think of a good spot in the dialog to put it without it being "less obvious/discoverable".

I couldn't think of a way to 100% guarantee a valid link without making the API call at **some** stage, unless we embed the check somewhere during the build process (which assumes the tag on GitHub exists **before** building). Alternative suggestions welcome.

**Pull request closed without merging**